pynitrokey/cli/nk3/secrets.py

```diff
@@ -17,6 +23,45 @@ def secrets(ctx: click.Context) -> None:
     pass
 
 
+def repeat_if_pin_needed(func) -> Callable:  # type: ignore[no-untyped-def]
+    """
+    Decorated function should have at least one argument,
+    of which the first one should be an instance of the SecretsApp. Otherwise, a RuntimeError is raised.
+    """
+
+    def wrapper(*args, **kwargs) -> None:  # type: ignore[no-untyped-def]
+        assert len(args) >= 1 and isinstance(
+            args[0], SecretsApp
+        ), "repeat_if_pin_needed: SecretsApp should be passed as an argument to this decorator"
+        app: SecretsApp = args[0]
+
+        try:
+            if app.protocol_v2_confirm_all_requests_with_pin():
+                authenticate_if_needed(app)
+            func(*args, **kwargs)
+        except SecretsAppException as e:
+            # Behavior below is for the v3 version of the protocol. Bail if v2 is used.
+            if app.protocol_v2_confirm_all_requests_with_pin():
+                raise
+
+            if e.to_id() not in {
+                SecretsAppExceptionID.SecurityStatusNotSatisfied,
+                SecretsAppExceptionID.NotFound,
+            }:
+                raise
+            if e.to_id() == SecretsAppExceptionID.SecurityStatusNotSatisfied:
+                local_print("PIN is required to run this command.")
+            elif e.to_id() == SecretsAppExceptionID.NotFound:
+                local_print(
+                    "Credential not found. Please provide PIN below to search in the PIN-protected database."
+                )
```

Comment:
This would be easier to read:

```
            if e.to_id() == SecretsAppExceptionID.SecurityStatusNotSatisfied:
                local_print("PIN is required to run this command.")
            elif e.to_id() == SecretsAppExceptionID.NotFound:
                local_print(
                    "Credential not found. Please provide PIN below to search in the PIN-protected database."
                )
            else:
                raise
```

Reply:
Done